Back-End-Node/server.js

```diff
@@ -49,72 +50,53 @@ app.get("/auth/redirect", async (req, res) => {
     });
 
-    //console.log("neded", userProfile);
-    // //console.log("User Data", userDataResponse);
     const existingUser = await pool.query(
-      "SELECT * FROM public.person WHERE email = $1",
+      "SELECT * FROM person WHERE slack_email = $1",
       [userProfile["profile"]["email"]]
     );
 
-    let avatar = userProfile["profile"]["image_original"];
-    let defaultRole = userProfile["profile"]["title"];
-    if (existingUser.rows[0] && existingUser.rows[0]["avatar"] !== avatar) {
-      await pool.query("UPDATE public.person SET avatar = $1 WHERE id = $2", [
-        avatar,
-        existingUser.rows[0]["id"],
-      ]);
-    }
-    // Update the default_role if it has changed
-    if (
-      existingUser.rows[0]["default_role"] !== "admin" &&
-      existingUser.rows[0]["default_role"] !== defaultRole
-    ) {
-      await pool.query(
-        "UPDATE public.person SET default_role = $1 WHERE id = $2",
-        [defaultRole, existingUser.rows[0]["id"]]
-      );
-    }
-
     let jwtToken = "";
 
     if (existingUser.rows.length > 0) {
-      //console.log(existingUser);
-      //Login Bussiness
       jwtToken = createToken(
         existingUser.rows[0]["id"],
-        existingUser.rows[0]["default_role"]
+        existingUser.rows[0]["slack_title"]  // TODO: weird that title = jwt role
       );
+
+      // TODO: update values if changed
+      // // Update avatar if it has changed
+      // if (existingUser.rows[0]["avatar"] !== userProfile["profile"]["image_original"]) {
+      //   await pool.query("UPDATE person SET avatar = $1 WHERE id = $2", [
+      //     userProfile["profile"]["image_original"],
+      //     existingUser.rows[0]["id"],
+      //   ]);
+      // }
     } else {
       // Insert the new user into the database
       var insertResult = await pool.query(
-        "INSERT INTO public.person (created_at, homecity, default_role, email, first_name, last_name, avatar) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING id",
+        "INSERT INTO person (slack_photo_link, slack_firstname, slack_lastname, slack_email, slack_title) VALUES ($1, $2, $3, $4, $5) RETURNING id, slack_title",
         [
-          new Date(),
-          "London",
-          userProfile["profile"]["title"],
-          userProfile["profile"]["email"],
+          userProfile["profile"]["image_original"],
           userProfile["profile"]["first_name"],
           userProfile["profile"]["last_name"],
-          avatar,
+          userProfile["profile"]["email"],
+          userProfile["profile"]["title"],
         ]
       );
-
-      //login bussiness
       jwtToken = createToken(
         insertResult.rows[0]["id"],
-        userProfile["profile"]["title"]
+        insertResult.rows[0]["slack_title"]  // TODO: weird that title = jwt role
       );
     }
 
-    res.redirect(`${backendUrl}/oauthdone?code=${jwtToken}`);
+    // redirect back to frontend so that it can run setSession with this token
```

Comment:
this one is front url, the name was only backednd url for slack which is localhost:443. we all named it together, when we were changing hard code details at the office.